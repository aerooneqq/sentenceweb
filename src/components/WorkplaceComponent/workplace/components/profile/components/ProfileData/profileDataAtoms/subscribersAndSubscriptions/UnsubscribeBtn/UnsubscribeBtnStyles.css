.unsubscribeBtn { 
    width: 80px;
    height: 30px;

    background: orange;

    border-style: solid;
    border-width: 1px;
    border-radius: 7px;
    border-color: orange;

    font-family: Arial;
    color: white;
}

.unsubscribeBtn:focus{ 
    outline: none;
}

.unsubscribeBtn:hover { 
    background: linear-gradient(-120deg, darkorange, orange)
}