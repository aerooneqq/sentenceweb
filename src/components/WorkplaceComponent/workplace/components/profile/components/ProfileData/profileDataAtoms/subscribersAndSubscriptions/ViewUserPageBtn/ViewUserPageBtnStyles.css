.viewUserPageBtn{ 
    width: 80px;
    height: 30px;

    background: gray;

    border-style: solid;
    border-radius: 7px;
    border-color: gray;
    border-width: 1px;
    
    margin-left: 10px;

    font-family: Arial;
    color: white;
}

.viewUserPageBtn:focus{ 
    outline: none;
}

.viewUserPageBtn:hover{ 
    background: linear-gradient(-120deg,#707070, #808080)
}