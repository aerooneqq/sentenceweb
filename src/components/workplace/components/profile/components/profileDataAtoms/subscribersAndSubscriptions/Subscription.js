import React, {Component, lazy} from "react";

//Styles
import "./styles/SubscriptionStyles.css";

//Components
const ViewUserPageBtn = lazy(() => import("./ViewUserPageBtn"));

export default class Subscription extends Component{ 
    constructor(props){ 
        super(props);
    }

    render() { 
        return( 
            <div className = "subscriptionsListContainer">
                <div className = "subscriptionsListUserPhoto">
                </div>

                <div className = "subscriptionsListUserInfo">
                    <div className = "subscriptionsListUserName">
                        Степанов Евгений
                    </div>
                    <div className = "subscriptionsListUserBirthDate">
                        30.05.2000
                    </div>
                    <div class = "subscriptionsListUserBtns">
                        <ViewUserPageBtn />
                    </div>
                </div>
            </div>
        )
    }
}