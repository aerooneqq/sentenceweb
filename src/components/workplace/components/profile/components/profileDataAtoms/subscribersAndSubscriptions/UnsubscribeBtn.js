import React, {Component} from "react";

//Styles
import "./styles/UnsubscribeBtnStyles.css";

export default class UnsubscribeBtn extends Component{ 
    constructor(props){ 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick(){ 
        this.props.deleteSubscription(this.props.userID);
    }

    render() { 
        return (
            <button className = "unsubscribeBtn"
                    onClick = {this.handleBtnClick}>
                Unsub
            </button>
        )
    }
}