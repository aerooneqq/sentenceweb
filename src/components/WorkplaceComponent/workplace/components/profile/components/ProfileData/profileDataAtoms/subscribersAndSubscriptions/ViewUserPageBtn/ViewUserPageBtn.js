import React, {Component} from "react";

//Styles
import "./ViewUserPageBtnStyles.css"

/**
 * PROPS LIST: 
 * 1) viewUserPage
 * 2) userID
 */
export default class ViewUserPageBtn extends Component { 

    constructor(props) { 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick() { 
        this.props.viewUserPage(this.props.userID);
    }

    render() { 
        return( 
            <button className = "viewUserPageBtn"
                    onClick = {this.handleBtnClick}>
                View
            </button>
        )
    }
}