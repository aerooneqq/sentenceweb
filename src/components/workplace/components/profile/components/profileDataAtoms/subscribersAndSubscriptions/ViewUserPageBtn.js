import React, {Component} from "react";

//Styles
import "./styles/ViewUserPageBtnStyles.css"

export default class ViewUserPageBtn extends Component{ 
    constructor(props){ 
        super(props);
    }

    render(){ 
        return( 
            <button className = "viewUserPageBtn">
                View
            </button>
        )
    }
}