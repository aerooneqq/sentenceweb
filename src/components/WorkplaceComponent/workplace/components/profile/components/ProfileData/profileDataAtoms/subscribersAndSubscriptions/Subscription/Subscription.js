import React, {Component, lazy} from "react";

//Styles
import "./SubscriptionStyles.css";

//Components
import ViewUserPageBtn from "../ViewUserPageBtn/ViewUserPageBtn";
import UnsubscribeBtn from "../UnsubscribeBtn/UnsubscribeBtn";

/**
 * PROPS LIST: 
 * 1) subscription: {userID, name, birthDate}
 * 2) deleteSubscription
 */
export default class Subscription extends Component { 

    render() { 
        let date = this.props.subscription.birthDate;
        date = date.substr(0, 4) + "-" + date.substr(5, 2) + "-" + date.substr(8, 2);

        return ( 
            <div className = "subscriptionsListContainer">
                <div className = "subscriptionsListUserPhoto">
                </div>

                <div className = "subscriptionsListUserInfo">
                    <div className = "subscriptionsListUserName">
                        {this.props.subscription.name}
                    </div>
                    <div className = "subscriptionsListUserBirthDate">
                        {date}
                    </div>
                    <div class = "subscriptionsListUserBtns">
                        <UnsubscribeBtn deleteSubscription = {this.props.deleteSubscription}
                                        userID = {this.props.subscription.userID}/>
                        <ViewUserPageBtn />
                    </div>
                </div>
            </div>
        )
    }
}