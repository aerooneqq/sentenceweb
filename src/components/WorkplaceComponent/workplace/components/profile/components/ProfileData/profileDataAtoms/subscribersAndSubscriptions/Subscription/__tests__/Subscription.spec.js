import React from "react"; 
import {shallow} from "enzyme"; 
import {shallowToJson} from "enzyme-to-json";
import Subscription from "../Subscription";

import ViewUserPageBtn from "../../ViewUserPageBtn/ViewUserPageBtn";
import UnsubscribeBtn from "../../UnsubscribeBtn/UnsubscribeBtn";


const props = { 
    subscription: { 
        name: "Aero", 
        userID: 1,
        birthDate: "30.05.2000T23131212"
    },
    deleteSubscription: () => {}
}

describe("Subscription", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<Subscription {...props}/>))).toMatchSnapshot();
    });

    it("should display the subscription information", () => { 
        const subscription = shallow(<Subscription {...props}/>);

        expect(subscription.find("div.subscriptionsListUserName").text()).toBe(props.subscription.name);

        let {subscription: {birthDate}} = props;
        birthDate = birthDate.substr(0, 4) + "-" + birthDate.substr(5, 2) + "-" + birthDate.substr(8, 2);

        expect(subscription.find("div.subscriptionsListUserBirthDate").text()).toBe(birthDate);
    });

    it("should contain btns", () => {
        const subscription = shallow(<Subscription {...props}/>);

        expect(subscription.find(ViewUserPageBtn).exists()).toBe(true);
        expect(subscription.find(UnsubscribeBtn).exists()).toBe(true);
    });
});