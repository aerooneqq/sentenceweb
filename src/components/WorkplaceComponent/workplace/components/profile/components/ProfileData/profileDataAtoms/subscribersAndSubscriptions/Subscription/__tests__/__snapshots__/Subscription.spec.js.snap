// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription should render correctly 1`] = `
<div
  className="subscriptionsListContainer"
>
  <div
    className="subscriptionsListUserPhoto"
  />
  <div
    className="subscriptionsListUserInfo"
  >
    <div
      className="subscriptionsListUserName"
    >
      Aero
    </div>
    <div
      className="subscriptionsListUserBirthDate"
    >
      30.0-.2-00
    </div>
    <div
      class="subscriptionsListUserBtns"
    >
      <UnsubscribeBtn
        deleteSubscription={[Function]}
        userID={1}
      />
      <ViewUserPageBtn />
    </div>
  </div>
</div>
`;
